src/ring_theory/witt_vector/defs.lean

```diff
@@ -164,3 +302,5 @@ lemma witt_neg_vars (n : ℕ) :
 witt_structure_int_vars _ _ _
 
 end witt_vector
+
+attribute [irreducible] witt_vector
```

Comment:
I think this is an excellent use of [irreducible], you've made the interface and now people should use it properly.